channel/consensus_channel/leader_channel.go

```diff
@@ -51,25 +55,19 @@ func (c *ConsensusChannel) IsProposedNext(g Guarantee) (bool, error) {
 // and generates and stores a SignedProposal in the queue, returning the
 // resulting SignedProposal
 //
-// Note: the TurnNum on add is ignored; the correct turn number is computed
-// and applied by c
 func (c *ConsensusChannel) Propose(proposal Proposal, sk []byte) (SignedProposal, error) {
 	if c.MyIndex != Leader {
 		return SignedProposal{}, ErrNotLeader
 	}
 
-	// TODO: the Propose API should be less confusing!
-	// Currently, the TurnNum is ignored, and Propose could easily
-	// return the same ChannelId that it's been passed
-	proposal.ChannelID = c.Id
-
+	if proposal.ChannelID != c.Id {
```

Comment:
I think it makes more sense to enforce the channelId is correct, instead of overwriting it.

Reply:
Agreed. Silently overwriting the input is not expected. A different API surface which didn't allow for a passed-in channelID would take responsibility for setting that ID, but that's not what this method is.